### PR TITLE
fix(gatsby): Add a generic error handler to child state machines

### PR DESCRIPTION
**Pull request by sidharthachatterjee**
The develop state machine has appropriate onError handlers. This adds a generic error handler to the child state machines that logs the error.

### CODE REVIEW

**Comment by wardpeet** [comment]
@sidharthachatterjee wouldn't it be better to use "xstate strict mode"? I think it bubbles up the onErrors to the parent machine. https://xstate.js.org/docs/guides/communication.html#invoking-promises

**Comment by sidharthachatterjee** [comment]
@wardpeet From my and @pieh investigation, strict mode errors the process out if the state doesn’t include an onError. 

However, that log only gets printed if NODE_ENV=production so it doesn’t print anything in Gatsby develop 😅

**Review by wardpeet — APPROVED** [review]
I don't see anything wrong with this, okay to merge @sidharthachatterjee ?

**Comment by ascorbic** [comment]
Rather than defining this as a function that you import and add to every state, it's best practice to add it to the `actions` for the machine, and specify it by string. This makes it more testable.

File: packages/gatsby/src/utils/generic-on-error.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import reporter from "gatsby-cli/lib/reporter"
+
+export function genericOnError(_ctx, event): void {
```
Comment (ascorbic) [inline packages/gatsby/src/utils/generic-on-error.ts:3]: You can use reporter from context
```suggestion
export function genericOnError({ reporter }, event): void {
```

File: packages/gatsby/src/utils/generic-on-error.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  const error = event.data
+  reporter.panicOnBuild(error)
```
Comment (ascorbic) [inline packages/gatsby/src/utils/generic-on-error.ts:5]: (It would need an optional chain here though)

Comment (sidharthachatterjee) [reply packages/gatsby/src/utils/generic-on-error.ts]: Done

**Comment by LekoArts** [comment]
Since this went stale I'll close this for now.

**Pull request closed without merging**